src/Fields/MultiValueCheckboxField.php

```diff
@@ -208,6 +208,25 @@ public function saveInto(DataObjectInterface $record)
         }
     }
 
+    /**
+     * Load the value from the dataobject into this field
+     *
+     * @param DataObject|DataObjectInterface $record
+     */
+    public function loadFrom(DataObjectInterface $record)
+    {
+        $fieldName = $this->getName();
+        if (empty($fieldName) || empty($record)) {
+            return;
+        }
+
+        if ($record->hasField($fieldName)) {
+            $value = $record->$fieldName;
```

Comment:
I would probably use `->getField($fieldName)` here since if there was a method with a name similar to a field, it would take precedence because of how the magical access works in DOs (I _think_). At the same time, this might be ok. What do you think @GuySartorelli?

Reply:
I think it's correct to respect getter methods and other model-specific overrides here. That's what `MultiSelectField` does too:
https://github.com/silverstripe/silverstripe-framework/blob/c278576996f73cf8be53924c0fe55dff86b5045e/src/Forms/MultiSelectField.php#L85-L113

I am a little concerned that there's a lot of functionality in the `MultiSelectField` implementation that isn't being used here - though I haven't spend much time looking at this PR or the problem it's fixing beyond finding what I need to answer the above question.

That said, the problem as described is about `stringDecode()` wanting a string when an array is passed, but that method seems like it handles arrays? Am I missing something here or are we fixing this in the wrong place?

Reply:
done.

Reply:
Yeah, I haven't looked too deep into this either and I think we did some custom stuff locally, so I have to have a look first. The fix might be something else somewhere else, so bear with us @xini.

Reply:
@michalkleiner As far as I can see the conversion from db field value to form field value seems to be the issue. I believe at some point json was added as a storage format additional to serialisation. And I believe that's what caused the error here? But I could be wrong.

Reply:
I think `->getField($fieldName)` is cleaner and safer. Only because `MultiSelectField` uses `$record->$fieldName` doesn't mean it's right. I know I copied it from `MultiSelectField`, but I think you're right about `->getField($fieldName)` being the better option.

Reply:
note respecting getter methods (i.e. using `getField()` explicitly) would probably be a breaking change.

Reply:
reverted.